xbmc/guilib/Texture.cpp

```diff
@@ -212,6 +212,7 @@ std::unique_ptr<CTexture> CTexture::LoadFromFile(const std::string& texturePath,
 
       std::unique_ptr<CTexture> texture = CTexture::CreateTexture();
       texture->LoadFromMemory(width, height, width*4, XB_FMT_RGBA8, true, inputBuff);
+      delete[] inputBuff;
```

Comment:
Better make on `unsigned char* inputBuff;` (L205) a `unsigned char* inputBuff = nullptr;` the `ReadIcon` can be return before it is set.

And yeah, stupid with memleaks as this :smirk:.

Reply:
Has forget to write, it can also return 0 if `malloc` already done on `ReadIcon`

Reply:
I changed `ReadIcon` so that the output parameters are always in a valid state if it returns with `!= 0` and not altered otherwise.